backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs

```diff
@@ -30,19 +30,33 @@ public async Task<UserDto> CreateUser(UserRegisterDto userDto)
 
         public async Task<LessonDto> EnrollUserToLesson(long lessonId)
         {
-            var user = _context.Users
-                .Single(p => p.Id == _firebaseAuthService.UserId);
+            var userId = _firebaseAuthService.UserId;
+            var user = _context.Users.FirstOrDefault(u => u.Id == userId);
 
-            Lesson existingLesson = _context.Lessons
-                .Single(p => p.Id == lessonId);
+            if (user == null)
+            {
+                throw new ArgumentException($"Failed to find the user with id {userId}");
+            }
 
-            user.Lessons.Add(existingLesson);
+            var lesson = _context.Lessons.FirstOrDefault(l => l.Id == lessonId);
+
+            if (lesson == null)
+            {
+                throw new ArgumentException($"Failed to find the lesson with id {lessonId}");
+            }
```

Comment:
var lesson = _context.Lessons.SingleOrDefault(l => l.Id == lessonId) ?? throw new ArgumentException($"Failed to load the lesson}");